### ISSUE TITLE
Cannot open a file with filename containing special characters.

### ISSUE THREAD

**Issue by katcom** [FolioReader/FolioReader-Android#261]
Issue / Feature - Issue 
FolioReader version - 0.4.1 
FolioReader Stock / Modified - Stock 
Android SDK - 26.1.1
Mobile / Tablet / Emulator Info - Xiaomi Mi-4c
Crash / Error - Error

Steps to reproduce / Describe in detail - 

I'm using library version 0.4.1  When trying to open epub from my Activity, eg -

```
FolioReader folioReader = FolioReader.get();
folioReader.openBook("file:///android_asset/Ebooks/Austen, Jane - Emma.epub"); 
```

Getting this error - (While i'm opening the file from my activity)

- `08-22 22:25:37.983 25576-2911/studio.ludwig.emulereader E/fi.iki.elonen.NanoHTTPD: Communication with the client broken, or an bug in the handler code
                                                                                   java.lang.NullPointerException: Attempt to invoke virtual method 'fi.iki.elonen.NanoHTTPD$Response fi.iki.elonen.router.RouterNanoHTTPD$UriResource.process(java.util.Map, fi.iki.elonen.NanoHTTPD$IHTTPSession)' on a null object reference
                                                                                       at fi.iki.elonen.router.RouterNanoHTTPD$UriRouter.process(RouterNanoHTTPD.java:596)
                                                                                       at fi.iki.elonen.router.RouterNanoHTTPD.serve(RouterNanoHTTPD.java:671)
                                                                                       at fi.iki.elonen.NanoHTTPD$HTTPSession.execute(NanoHTTPD.java:945)
                                                                                       at fi.iki.elonen.NanoHTTPD$ClientHandler.run(NanoHTTPD.java:192)
                                                                                       at java.lang.Thread.run(Thread.java:818)
08-22 22:25:37.984 25576-25846/studio.ludwig.emulereader E/ManifestTask: ManifestTask failed
                                                                         java.io.IOException: unexpected end of stream on Connection{127.0.0.1:8080, proxy=DIRECT@ hostAddress=127.0.0.1 cipherSuite=none protocol=http/1.1} (recycle count=0)
                                                                             at com.android.okhttp.internal.http.HttpConnection.readResponse(HttpConnection.java:210)
                                                                             at com.android.okhttp.internal.http.HttpTransport.readResponseHeaders(HttpTransport.java:80)
                                                                             at com.android.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:904)
                                                                             at com.android.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:788)
                                                                             at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:443)
                                                                             at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:388)
                                                                             at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:231)
                                                                             at com.folioreader.ui.base.ManifestTask.doInBackground(ManifestTask.java:43)
                                                                             at com.folioreader.ui.base.ManifestTask.doInBackground(ManifestTask.java:26)
                                                                             at android.os.AsyncTask$2.call(AsyncTask.java:295)
                                                                             at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                             at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
                                                                             at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                                                                             at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                                                                             at java.lang.Thread.run(Thread.java:818)
                                                                          Caused by: java.io.EOFException: \n not found: size=0 content=...
                                                                             at com.android.okhttp.okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:200)
                                                                             at com.android.okhttp.internal.http.HttpConnection.readResponse(HttpConnection.java:191)
                                                                             at com.android.okhttp.internal.http.HttpTransport.readResponseHeaders(HttpTransport.java:80) 
                                                                             at com.android.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:904) 
                                                                             at com.android.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:788) 
                                                                             at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:443) 
                                                                             at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:388) 
                                                                             at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:231) 
                                                                             at com.folioreader.ui.base.ManifestTask.doInBackground(ManifestTask.java:43) 
                                                                             at com.folioreader.ui.base.ManifestTask.doInBackground(ManifestTask.java:26) 
                                                                             at android.os.AsyncTask$2.call(AsyncTask.java:295) 
                                                                             at java.util.concurrent.FutureTask.run(FutureTask.java:237) 
                                                                             at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234) 
                                                                             at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
                                                                             at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
                                                                             at java.lang.Thread.run(Thread.java:818) `

**Comment by hrishikesh-kadam** [comment]
@katcom Thanks for filling the issue template. I think you mean to say 5.1.1 instead of 26.1.1 in Android SDK.

I am unable to reproduce the issue. Can you please send us the EPUB for further investigation?

**Comment by katcom** [comment]
Simply wired. There is the EPUB I had used.

[Austen, Jane - Emma.zip](https://github.com/FolioReader/FolioReader-Android/files/2328598/Austen.Jane.-.Emma.zip)

**Comment by katcom** [comment]
Thanks :)

**Comment by hrishikesh-kadam** [comment]
@katcom Issue fix released in v0.4.2
